packages/libs/hd-wallet/src/index.ts

```diff
@@ -1 +1,257 @@
+  }
+  if (oldPassword === newPassword) {
+    throw new Error(
+      'The new password must be different from the old password.',
+    );
+  }
+
+  let decryptedPrivateKey;
+  try {
+    decryptedPrivateKey = decryptPrivateKey(privateKey, oldPassword);
+  } catch (error) {
+    throw new Error(
+      `Failed to decrypt the private key with the old password: ${error.message}`,
+    );
+  }
+
+  let newEncryptedPrivateKey;
+  try {
+    newEncryptedPrivateKey = encryptPrivateKey(
+      decryptedPrivateKey,
+      newPassword,
+    );
+  } catch (error) {
+    throw new Error(
+      `Failed to encrypt the private key with the new password: ${error.message}`,
+    );
+  }
+
+  return newEncryptedPrivateKey;
+}
+
+/**
+ * Convert a given mnemonic phrase into a seed buffer.
+ *
+ * @param {string} mnemonic - A mnemonic seed phrase to be converted into a seed buffer.
+ * @param {string} [password] - Optional password for encrypting the seed.
+ * @throws {Error} Throws an error if the provided mnemonic is not valid.
+ * @returns {Promise<{ seedBuffer: Uint8Array, seed: string }>} - Returns the seed buffer and processed seed.
+ */
+export async function kadenaGenSeedFromMnemonic(
+  mnemonic: string,
+  password?: string,
+): Promise<{ seedBuffer: Uint8Array; seed: string }> {
+  if (bip39.validateMnemonic(mnemonic, wordlist) === false) {
+    throw Error('Invalid mnemonic.');
+  }
+
+  const seedBuffer = await bip39.mnemonicToSeed(mnemonic);
+  let seed: string;
+
+  if (typeof password === 'string' && password.length > 0) {
+    seed = encryptOrEncodeSeed(seedBuffer, password);
+  } else {
+    seed = encryptOrEncodeSeed(seedBuffer);
+  }
+  return {
+    seedBuffer,
+    seed,
+  };
+}
+
+/**
+ * Generates a mnemonic phrase using the BIP39 protocol with a specified wordlist.
+ *
+ * @returns {string} A valid BIP39 mnemonic phrase.
+ * @throws {Error} If the generated mnemonic is invalid.
+ */
+export function kadenaGenMnemonic(): string {
+  const words = bip39.generateMnemonic(wordlist);
+  if (bip39.validateMnemonic(words, wordlist) === false) {
+    throw Error('Invalid mnemonic.');
+  }
+  return words;
+}
+
+/**
+ * Generates a key pair from a seed buffer and an index or range of indices, and optionally encrypts the private key.
+ *
+ * @param {Uint8Array} seedBuffer - The seed buffer to use for key generation.
+ * @param {number | [number, number]} indexOrRange - Either a single index or a tuple with start and end indices for key pair generation.
+ * @param {string} [password] - Optional password for encrypting the private key.
+ * @returns {([string, string] | [string, string][])} - Depending on the input, either a tuple for a single key pair or an array of tuples for a range of key pairs, with the private key encrypted if a password is provided.
+ * @throws {Error} Throws an error if the seed buffer is not provided, if the indices are invalid, or if encryption fails.
+ */
+export function kadenaGenKeypair(
+  seedBuffer: Uint8Array,
```

Comment:
I think we should accept the encrypted seed instead. 
we can decrypt it easily with the password
What do you say?